Code/DataStructs/BitOps.cpp

```diff
@@ -834,14 +835,14 @@ double CalcBitmapTanimoto(const unsigned char* afp, const unsigned char* bfp,
     intersect_popcount += byte_popcounts[afp[i] & bfp[i]];
   }
 #else
-  unsigned int eidx = nBytes / sizeof(unsigned int);
-  for (unsigned int i = 0; i < eidx; ++i) {
-    union_popcount +=
-        __builtin_popcount(((unsigned int*)afp)[i] | ((unsigned int*)bfp)[i]);
-    intersect_popcount +=
-        __builtin_popcount(((unsigned int*)afp)[i] & ((unsigned int*)bfp)[i]);
-  }
-  for (unsigned int i = eidx * sizeof(unsigned int); i < nBytes; ++i) {
+  boost::uint64_t eidx = nBytes / sizeof(boost::uint64_t);
+  for (boost::uint64_t i = 0; i < eidx; ++i) {
+    union_popcount += __builtin_popcountl(((boost::uint64_t*)afp)[i] |
```

Comment:
Good catch there. I tend to start with the C-style cast and then go back and convert them to something more explicit.